src/web3/WalletClient.ts

```diff
@@ -377,6 +378,39 @@ export class WalletClient extends BaseClient implements IWalletClient {
     } as ISignature;
   }
 
+  /**
+   * Verify a signature.
+   *
+   * @param data The signed data to verify
+   * @param signature The signature to verify
+   * @param signerPubKey The public key of the signer
+   * @returns true if the signature is valid, false otherwise
+   */
+  public async verifySignature(
+    data: string | Buffer,
+    signature: ISignature,
+    signerPubKey: string,
+  ): Promise<boolean> {
+    // setup the public key
+    const publicKeyBase58Decoded = WalletClient.getBytesPublicKey(signerPubKey);
+
+    // setup the message digest
+    const bytesCompact: Buffer = Buffer.from(data);
+    const messageHashDigest: Uint8Array = hashBlake3(bytesCompact);
+
+    // setup the signature
+    const hexSignature = base58Decode(signature.base58Encoded);
```

Comment:
```suggestion
    const signatureBytes = base58Decode(signature.base58Encoded);
```

Reply:
Why not setting a type? this is not detected by the linter in the CI ?

Reply:
looks like it is not detected.